portfolio/src/main/webapp/charts.js

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+google.charts.load('current', { 
+    'packages':['corechart', 'bar', 'geochart'],
+    'mapsApiKey': 'AIzaSyAocmGfmH_sqksYwwYlEOJ_5jjhneLDCxg'
+    });
+google.charts.setOnLoadCallback(drawCharts);
+
+/** Fetches covid data and uses it to create a chart. */
+function drawCharts() {
+  fetch('/get-covidData').then(response => response.json())
+  .then((covidData) => {
+    createChartBySex(covidData['covidBySex']);
+    createChartByMunicipality(covidData['covidByMunicipality']);
+    createChartByAgeRange(covidData['covidByAgeRange']);
+    createChartByState(covidData['deathsByState']);
+  });
+}
+
+/**
+ * Create a donut chart, showing cases by sex.
+ * @param {Object<string, number>} covidData 
+ */
+function createChartBySex(covidData) {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Sex');
+  data.addColumn('number', 'Covid-19 Cases');
+
+  data.addRows([
+    ['Women', covidData['M']],
+    ['Men', covidData['H']]
+  ]);
+
+  const options = {
+    chartArea: {
+      height: '80%',
+      width: '80%'
+    },
+    legend: {alignment: 'center', position: 'bottom'},
+    title: 'Covid-19 Cases: Women and Men',
+    pieHole: 0.4,
+    width: 500,
+    height: 400
+  };
```

Comment:
It looks like each chart has slightly different options, so it probably makes them hard to reuse which is unfortunate. But in either case, I'd consider moving these out of the methods and make them top-level constants, or put them into a separate file/library, like ChartConfigs.js or something.

It will make the logic of the methods easier to parse.